FocusOn/FocusOnApp.swift

```diff
@@ -9,12 +9,18 @@ import SwiftUI
 
 @main
 struct FocusOnApp: App {
-    let persistenceController = PersistenceController.shared
+    let dataService = GoalDataService()
 
     var body: some Scene {
         WindowGroup {
             MainView()
-                .environment(\.managedObjectContext, persistenceController.container.viewContext)
+                .environment(\.managedObjectContext, dataService.container.viewContext)
```

Comment:
Your data service should exposed such information, I shouldn't be able to guess hat you are using CoreData.